src/main/java/com/datadoghq/sketch/ddsketch/mapping/IndexMappingConverter.java

```diff
@@ -0,0 +1,107 @@
+/* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+package com.datadoghq.sketch.ddsketch.mapping;
+
+import com.datadoghq.sketch.ddsketch.store.Bin;
+import com.datadoghq.sketch.ddsketch.store.BinAcceptor;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * An interface for converting bins that have been encoded using an {@link IndexMapping} to bins
+ * encoded using another one.
+ */
+public interface IndexMappingConverter {
+
+  /**
+   * Converts bins.
+   *
+   * @param inBins an ascending iterator, that is, an iterator that returns bins whose indexes are
+   *     sorted in ascending order
+   * @param outBins a consumer that is fed the converted bins
+   * @throws IllegalArgumentException if the provided iterator is not ascending
+   */
+  void convertAscendingIterator(Iterator<Bin> inBins, BinAcceptor outBins);
+
+  /**
+   * Returns a converter that uniformly distributes the count of a bin to the overlapping bins of
+   * the new mapping based on the shares of the initial bin that the new bins cover.
+   *
+   * <p>This conversion method is not the one that minimizes the relative accuracy of the quantiles
+   * that are computed from the resulting bins. For instance, transferring the full count of a bin
+   * of the initial mapping to the single bin of the new mapping that overlaps {@link
+   * IndexMapping#value(int)} of the initial mapping allows computing more accurate quantiles.
+   * However, this method produces better-looking histograms and avoids conversion artifacts that
+   * would cause empty bins or bins with counts that are excessively high relative to its
+   * neighbors'.
+   *
+   * <p>If \(\alpha_i\) is the relative accuracy of the initial mapping {@code inMapping},
+   * \(\alpha_o\) the relative accuracy of the new mapping {@code outMapping}, and assuming that the
+   * initial bins are not themselves resulting from a conversion (in which case \(\alpha_i\) needs
+   * to be adjusted to be the effective relative accuracy of the initial bins), the effective
+   * relative accuracy of the quantiles that are computed from the bins that result from this
+   * conversion method is upper-bounded by \(\alpha = \frac{\gamma-1}{\gamma+1}\) where \(\gamma =
+   * \gamma_i^2\gamma_o\), \(\gamma_i = \frac{1+\alpha_i}{1-\alpha_i}\), and \(\gamma_o =
```

Comment:
where does \gamma_i^2\gamma_o come from?

Reply:
I'll add a comment, but essentially, that's because that method causes the count of a single point to be spread over the full width of a bin of the initial mapping, causing up to `~2*\alpha_i ` relative error (if relative accuracies are small). In addition, the allocation of counts to the bins of the new mapping causes `\alpha_o` relative error.

For instance, if we have a single point, we could have:
```
single input value:                                      x
initial mapping:                       -|-------o-------|-------o-------|-------o-------|
max (q_1) after bin encoding (1):                               x
count spreading over full bin (2):                      [---------------]
new mapping:                           |---o---|---o---|---o---|---o---|---o---|---o---|-
non-empty bins after conversion (3):                       o       o       o
max after conversion:                                                      x
```

The resulting max (i.e., value at quantile 1) is shifted by `\alpha_i` because of (1), an additional `\alpha_i` because of (2) and `\alpha_o` because of (3).

In the non-approximate geometric domain, that is `\sqrt{\gamma} = \gamma_i\sqrt{\gamma_o}`

Reply:
I think my calculation wasn't correct (the approximation still was). I updated it, as well as the explanation, in 42003628c878d6bfd6e13457ca533bda084a3d63. Let me know if it makes more sense now.

![image](https://user-images.githubusercontent.com/10637860/121926620-6caee600-cd3e-11eb-8cfc-f0c14c45e590.png)

Reply:
Done in 36aadaf7ae6c935ed6e040d523588d386a53587a.